spyder/plugins/variableexplorer/widgets/namespacebrowser.py

```diff
@@ -284,6 +288,18 @@ def get_view_settings(self):
             settings[name] = getattr(self, name)
         return settings
 
+    def search(self):
+        """Handle showing/hiding search widget."""
+        self.editor.finder.setText('')
+        state = not self.finder.isVisible()
+        self.finder.setVisible(state)
+        self.search_button.setChecked(state)

```

Comment:
Please move these four lines to a new method called `show_finder(self, state)`, with `state=True` showing the finder and `state=False` hiding it.

Reply:
Also, it's a bit confusing that we have here `self.editor.finder` and `self.finder`, although I understand that `self.finder` is the widget that holds the search field and the button to its left. So let's remove `self.editor.finder` and instead let's attach `text_finder` to `finder`, like this:

```python
text_finder = NamespacesBrowserFinder(self.editor,
                                      callback=self.editor.set_regex,
                                      main=self,
                                      regex_base=VALID_VARIABLE_CHARS)
...
self.finder.text_finder = text_finder
```